### ISSUE TITLE
Segmentation fault in zero_grad()

### ISSUE THREAD

**Issue by bmwshop** [pytorch/pytorch#75470]
### 🐛 Describe the bug

An attempt to run this in Colab or docker, etc. on a GPU fails due a segmentation fault (see trace below)

https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_with_Subword_Tokenization.ipynb

Python 3.8.12
torch: '1.12.0a0+2c916ef'
pl: '1.6.0'

The same code appears to work in native mixed precision

### Versions

Epoch 0:   6%|██████▋                                                                                                               | 2/35 [00:00<00:07,  4.38it/s, loss=735, v_num=15]Fatal Python error: Segmentation fault

Thread 0x00007f2497fff700 (most recent call first):
<no Python frame>

Thread 0x00007f257ffff700 (most recent call first):
  File "/opt/conda/lib/python3.8/threading.py", line 306 in wait
  File "/opt/conda/lib/python3.8/threading.py", line 558 in wait
  File "/opt/conda/lib/python3.8/site-packages/tqdm/_monitor.py", line 60 in run
  File "/opt/conda/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/opt/conda/lib/python3.8/threading.py", line 890 in _bootstrap

Thread 0x00007f258bfff700 (most recent call first):
  File "/opt/conda/lib/python3.8/threading.py", line 306 in wait
  File "/opt/conda/lib/python3.8/queue.py", line 179 in get
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/summary/writer/event_file_writer.py", line 227 in run
  File "/opt/conda/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/opt/conda/lib/python3.8/threading.py", line 890 in _bootstrap

Current thread 0x00007f27184de740 (most recent call first):
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/optimizer.py", line 217 in zero_grad
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/core/lightning.py", line 1648 in optimizer_zero_grad
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1596 in _call_lightning_module_hook
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 403 in _optimizer_zero_grad
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 297 in zero_grad_fn
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 140 in closure
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 148 in __call__
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/plugins/precision/precision_plugin.py", line 140 in _wrap_closure
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/adam.py", line 112 in step
  File "/opt/conda/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27 in decorate_context
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/optimizer.py", line 88 in wrapper
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/lr_scheduler.py", line 65 in wrapper
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/plugins/precision/precision_plugin.py", line 155 in optimizer_step
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/strategies/strategy.py", line 193 in optimizer_step
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/core/optimizer.py", line 168 in step
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/core/lightning.py", line 1625 in optimizer_step
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1596 in _call_lightning_module_hook
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 369 in _optimizer_step
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 256 in _run_optimization
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/optimization/optimizer_loop.py", line 203 in advance
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/base.py", line 204 in run
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/batch/training_batch_loop.py", line 88 in advance
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/base.py", line 204 in run
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py", line 208 in advance
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/base.py", line 204 in run
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/fit_loop.py", line 269 in advance
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loops/base.py", line 204 in run
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1354 in _run_train
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1324 in _run_stage
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1237 in _run
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 812 in _fit_impl
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 724 in _call_and_handle_interrupt
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 771 in fit
  File "t.py", line 89 in <module>
Segmentation fault (core dumped)

**Comment by duskvirkus** [comment]
I can attempt to replicate this weekend and see if I can narrow down the problem.

**Comment by bmwshop** [comment]
withdrawn; this does not appear to be an issue with pytorch.

**Comment by montanier** [comment]
> withdrawn; this does not appear to be an issue with pytorch.

Hello, could you give a bit more hints on the the issue and how you solved it on your end? I have a segfault on exactly the same line.

**Comment by lefreud** [comment]
I have the same issue (exact same trace as you) while training using the following libraries:
```
torch==1.10.0+cu113
pytorch-lightning==1.2.9
torchvision==0.11.1
```
And Python 3.8.10. @bmwshop do you have an idea where this issue could come from? Could it be an issue with CPython?

**Comment by schen207** [comment]
I'm facing the same issue, this is my environment:
```
pytorch-lightning==1.5.10
torch==1.11.0+cu113
Python==3.8.13
```

@bmwshop any insight on how you fixed the issue or where it came from would be much appreciated!
